### PR TITLE
[Optimization engine] Added troubleshooting page and split FAQ

### PR DESCRIPTION
**Pull request by helderpinto**
## 🛠️ Description

Added troubleshooting page and split FAQ in order to provide more information to customers facing problems that can be solved without opening an issue.

### 🔬 How did you test this change?

> - [ ] 🤏 Lint tests
> - [ ] 🤞 PS -WhatIf / az validate
> - [X] 👍 Manually deployed + verified
> - [ ] 💪 Unit tests
> - [ ] 🙌 Integration tests

### 🙋‍♀️ Do any of the following that apply?

> - [ ] 🚨 This is a breaking change.
> - [ ] 🤏 The change is less than 20 lines of code.

### 📑 Did you update `docs/changelog.md`?

> - [X] ✅ Updated changelog (required for `dev` PRs)
> - [ ] ➡️ Will add log in a future PR (feature branch PRs only)
> - [ ] ❎ Log not needed (small/internal change)

### 📖 Did you update documentation?

> - [X] ✅ Public docs in `docs` (required for `dev`)
> - [ ] ✅ Internal dev docs in `src` (required for `dev`)
> - [ ] ➡️ Will add docs in a future PR (feature branch PRs only)
> - [ ] ❎ Docs not needed (small/internal change)

### CODE REVIEW

File: docs/_optimize/optimization-engine/configuring-workspaces.md [hunk 1]
```diff
@@ -50,32 +50,9 @@ Install-Module -Name Az.OperationalInsights
 ./Setup-DataCollectionRules.ps1 -DestinationWorkspaceResourceId "/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/myResourceGroup/providers/Microsoft.OperationalInsights/workspaces/myWorkspace" -IntervalSeconds 30 -ResourceTags @{"tagName"="tagValue";"otherTagName"="otherTagValue"}
 ```
 
-### Log Analytics agent (legacy Microsoft Monitoring Agent)
+### Log Analytics agent (legacy Microsoft Monitoring Agent, deprecated on August 31st, 2024)
```
Comment (flanakin) [inline docs/_optimize/optimization-engine/configuring-workspaces.md:53]:
```suggestion
### Log Analytics agent (legacy Microsoft Monitoring Agent, deprecated on August 31, 2024)
```

File: docs/_optimize/optimization-engine/configuring-workspaces.md [hunk 2]
```diff
@@ -50,32 +50,9 @@ Install-Module -Name Az.OperationalInsights
...
-# Example 1 - just check all the workspaces configuration
-./Setup-LogAnalyticsWorkspaces.ps1
-
-# Example 2 - fix all workspaces configuration (using default counter collection frequency)
-./Setup-LogAnalyticsWorkspaces.ps1 -AutoFix
-
-# Example 3 - fix specific workspaces configuration, using a custom counter collection frequency
-./Setup-LogAnalyticsWorkspaces.ps1 -AutoFix -WorkspaceIds "d69e840a-2890-4451-b63c-bcfc5580b90f","961550b2-2c4a-481a-9559-ddf53de4b455" -IntervalSeconds 30
-```
+If you are still using the legacy Log Analytics agent, please migrate to the [Azure Monitor Agent](https://learn.microsoft.com/azure/azure-monitor/agents/azure-monitor-agent-migration).
```
Comment (flanakin) [inline docs/_optimize/optimization-engine/configuring-workspaces.md:55]: nit: Do we need or want to call out when we'll drop support for the Log analytics agent? If it doesn't impact our customers, then no worries. If it does, we should add something to the changelog that it's deprecated now and call out when we'll completely remove support for it. If you look at the changelog you'll see an instance of this with FOCUS PowerShell commands which were "deprecated" first, then "removed" later.

(FYI - Removing the setup instructions now is totally fine, even if we still have support for it.)

File: docs/_resources/changelog.md [hunk 3]
```diff
@@ -98,6 +98,10 @@ Legend:
 🔍 Optimization engine
 {: .fs-5 .fw-500 .mt-4 mb-0 }
 
+> ➕ Added:
+>
+> 1. Troubleshooting documentation page with the most common deployment and runtime issues and respective solutions or troubleshooting steps.
```
Comment (flanakin) [inline docs/_resources/changelog.md:103]:
```suggestion
> 1. [Troubleshooting documentation page](../_optimize/optimization-engine/troubleshooting.md) with the most common deployment and runtime issues and respective solutions or troubleshooting steps.
```

File: docs/_resources/changelog.md [hunk 4]
```diff
@@ -98,6 +98,10 @@ Legend:
 🔍 Optimization engine
```
Comment (flanakin) [inline docs/_resources/changelog.md:98]: Should we call out that we are deprecating our support of the Log analytics agent in a future release?

**Review by flanakin — APPROVED** [review]

File: docs/_optimize/optimization-engine/configuring-workspaces.md (hunk 2 above)
Comment (helderpinto) [reply docs/_optimize/optimization-engine/configuring-workspaces.md:55]: I agree we should add a note in the changelog. I'll commit an additional changelog entry.

File: docs/_resources/changelog.md (hunk 4 above)
Comment (helderpinto) [reply docs/_resources/changelog.md:98]: Agree. See latest commit.

**Pull request merged**